packages/components/src/Interactive.ts

```diff
@@ -35,124 +35,133 @@ const allowedHoverVariables = [
-      disabled,
-      hover = {},
-      role,
-      style = {},
-      tabIndex,
-      tagName: TagName = 'div',
-      type,
-      onBlur,
+      ref,
+      className: `terra-ui--interactive ${className}`,
+      style: {
+        ...style,
+        ...cssVariables,
+        fill: `var(--hover-color, ${style.color})`,
+        cursor: computedCursor,
+        outline,
+      },
+      role: computedRole,
+      tabIndex: computedTabIndex,
       onClick,
-      onKeyDown,
-      onMouseDown,
-      ...props
-    },
-    ref
-  ) => {
-    const [outline, setOutline] = useState<string>();
-    const { cursor } = style;
-
-    const computedCursor = Utils.cond(
-      [!!cursor, () => cursor],
-      [disabled, () => 'not-allowed'],
-      [!!onClick || pointerTags.includes(TagName) || pointerTypes.includes(type!), () => 'pointer']
-    );
-
-    const computedTabIndex = Utils.cond(
-      [_.isNumber(tabIndex), () => tabIndex],
-      [disabled, () => -1],
-      [!!onClick, () => 0],
-      () => undefined
-    );
-
-    const computedRole = Utils.cond(
-      [!!role, () => role],
-      [onClick && !['input', ...pointerTags].includes(TagName), () => 'button'],
-      () => undefined
-    );
-
-    const cssVariables = _.flow(
-      _.toPairs,
-      _.flatMap(([key, value]) => {
-        console.assert(
-          allowedHoverVariables.includes(key),
-          `${key} needs to be added to the .terra-ui--interactive CSS for the style to be applied`
-        );
-        return [
-          [`--app-hover-${key}`, value],
-          [key, `var(--hover-${key}, ${style[key]})`],
-        ];
-      }),
-      _.fromPairs
-    )(hover);
-
-    return createElement(
-      TagName,
-      {
-        ref,
-        className: `terra-ui--interactive ${className}`,
-        style: {
-          ...style,
-          ...cssVariables,
-          fill: `var(--hover-color, ${style.color})`,
-          cursor: computedCursor,
-          outline,
-        },
-        role: computedRole,
-        tabIndex: computedTabIndex,
-        onClick,
-        disabled,
-        onMouseDown: (e) => {
-          setOutline('none');
-          if (onMouseDown) {
-            onMouseDown(e);
-          }
-        },
-        onBlur: (e) => {
-          if (outline) {
-            setOutline(undefined);
-          }
-          if (onBlur) {
-            onBlur(e);
-          }
-        },
-        onKeyDown:
-          onKeyDown ||
-          ((event: React.KeyboardEvent) => {
-            if (event.key === 'Enter') {
-              event.stopPropagation();
-              (event.target as HTMLElement).click();
-            }
-          }),
-        ...props,
+      disabled,
+      onMouseDown: (e) => {
+        setOutline('none');
+        if (onMouseDown) {
+          onMouseDown(e);
+        }
+      },
+      onBlur: (e) => {
+        if (outline) {
+          setOutline(undefined);
+        }
+        if (onBlur) {
+          onBlur(e);
+        }
```

Comment:
I like optional chaining here to convey the full sentiment ("call this if its defined") in one shot:
```suggestion
        onBlur?.(e);
```

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4255/commits/8b7ba35c8b4280c6f50d835b483cb34fbd3e14b5.

Reply:
not gonna stop the train, but we should be careful with optional chaining since it can create unit testing coverage blindspot.

Reply:
On that note, unit tests for Interactive to come up in a future PR.

I'm also wondering if we may be able to get rid of these event handlers (which are used by modifying the `outline` style) with CSS based on [`:focus-visible`](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible).